### PR TITLE
Turn insertEpics into getEpics

### PR DESCRIPTION
**Pull request by maxspencer**
This PR turns the `insertEpics` function from the `Webview` service into a `getEpics` function in the 'Native' interface.

`getEpics` should return a non-empty list of epics to be displayed, or an empty list if no epics should be displayed.

This PR is to support issue #12

### CODE REVIEW

**Comment by maxspencer** [comment]
@JamieB-gu @webb04 what do you think about this?

**Comment by webb04** [comment]
I think it makes sense to have the majority of communication webview -> native.
It seems a shame that the font resizing might have to be the only native -> webview we've identified so far.

**Comment by webb04** [comment]
Are we still happy with returning a list of epics? I guess that's for possible future versions supporting multiple epics?

**Comment by maxspencer** [comment]
> Are we still happy with returning a list of epics? I guess that's for possible future versions supporting multiple epics?

I decided to leave that the same, but I was kinda thinking it might be wise to make it only return one. Epics seem quite solitary things, I don't know why we would want to display more than one?

**Comment by alexanderedge** [comment]
> > Are we still happy with returning a list of epics? I guess that's for possible future versions supporting multiple epics?
> 
> I decided to leave that the same, but I was kinda thinking it might be wise to make it only return one. Epics seem quite solitary things, I don't know why we would want to display more than one?

If `getEpic()` returns an optional `Epic`, the API could be extended to support multiple epics fairly easily.

**Comment by maxspencer** [comment]
> If `getEpic()` returns an optional `Epic`, the API could be extended to support multiple epics fairly easily.

Sounds sensible, but it doesn't seem like there's an easy way to have optional return types in Thrift definitions, other than by using an empty list, so maybe we should leave it as it is after all?

**Comment by maxspencer** [comment]
I'd be glad to be corrected on that though

**Comment by webb04** [comment]
Is this ok to be merged? I've implemented the current `insertEpics` but I can refactor it to use `getEpics` before merging https://github.com/guardian/apps-rendering/pull/195

File: thrift/native.thrift [hunk 1]
```diff
@@ -25,4 +32,6 @@ service Native {
     void launchSlideshow(1:list<Image> images, 2:i32 selectedIndex),
     bool isFollowing(1:Topic topic),
     bool isPremiumUser(),
+    void launchSlideshow(1:list<Image> images, 2:i32 selectedIndex),
```
Comment (alexanderedge) [inline thrift/native.thrift]: Should this be in here twice?

**Comment by davidfurey** [comment]
I can't find an optional return type either, although you could define:
```
struct MaybeEpic {
  1: optional Epic epic
}
```
and implement `MaybeEpic getEpic()`

File: thrift/native.thrift (hunk 1 above)
Comment (maxspencer) [reply thrift/native.thrift]: No, must have introduced this by mistake when merging.

**Comment by maxspencer** [comment]
> I can't find an optional return type either, although you could define:
> 
> ```
> struct MaybeEpic {
>   1: optional Epic epic
> }
> ```
> 
> and implement `MaybeEpic getEpic()`

That sounds OK to me, @JamieB-gu @webb04 will this be cool from the TypeScript side?

**Comment by webb04** [comment]
yeah we can add a check to see if we get an epic or not

**Comment by maxspencer** [comment]
Awesome, I'll implement your suggestion David

**Pull request merged**